### PR TITLE
[WIP] Feat: Environments

### PR DESCRIPTION
**Pull request by axe312ger**
## Summary

This will add environment support to the CLI

## Planned commands

* List
* Create
* Delete

## Todos

* [x] add list command
* [x] add create command
* [x] add delete command
* [ ] add tests

### CODE REVIEW

File: lib/cmds/space_cmds/environment_cmds/list.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      'https://github.com/contentful/contentful-cli/tree/master/docs/space/environment/list',
+      'Copyright 2018 Contentful, this is a BETA release'
+    ].join('\n'))
+}
+
+export const aliases = ['ls']
+
+async function environmentList (argv) {
+  await assertLoggedIn()
+  await assertSpaceIdProvided(argv)
```
Comment (Khaledgarbaya) [inline lib/cmds/space_cmds/environment_cmds/list.js]: Maybe at some point, we need to use [Middleware](https://github.com/yargs/yargs/blob/master/docs/advanced.md#middleware)

Comment (axe312ger) [reply lib/cmds/space_cmds/environment_cmds/list.js]: yeah till then lets keep the old, not that ideal but working, pattern

File: lib/cmds/space_cmds/environment.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const command = 'environment'
+
+export const desc = 'Manage and list your environments'
```
Comment (Khaledgarbaya) [inline lib/cmds/space_cmds/environment.js]: I would remove the `your` part `Manage and list environments`

File: lib/cmds/space_cmds/environment_cmds/create.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+export const desc = 'Create an environment'
+
+export const builder = (yargs) => {
+  return yargs
+    .usage('Usage: contentful environment create --name \'Your Environment Name\'')
+    .option('environment-id', {
+      alias: 'e',
+      describe: 'Id of the environment to create',
+      demandOption: true
```
Comment (Khaledgarbaya) [inline lib/cmds/space_cmds/environment_cmds/create.js]: Why is the id required? we can still create an environment with the base62 id, oder?

File: lib/cmds/space_cmds/environment_cmds/create.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    })
+    .option('name', {
+      alias: 'n',
+      describe: 'Name of the environment to create'
+    })
+    .option('space-id', {
+      describe: 'ID of the space the environment will belong to',
```
Comment (Khaledgarbaya) [inline lib/cmds/space_cmds/environment_cmds/create.js]: the sentence is correct but `ID of the space **that** the environment will belong to` is better

File: lib/cmds/space_cmds/environment_cmds/delete.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import { createManagementClient } from '../../../utils/contentful-clients'
+import { assertLoggedIn, assertSpaceIdProvided } from '../../../utils/assertions'
+
+export const command = 'delete'
+
+export const desc = 'Delete an environment'
+
+export const builder = (yargs) => {
+  return yargs
+    .usage('Usage: contentful environment delete --environment-id \'staging\'')
```
Comment (Khaledgarbaya) [inline lib/cmds/space_cmds/environment_cmds/delete.js]: `contentful **space** environment delete --environment-id` ?

File: lib/cmds/space_cmds/environment_cmds/create.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import { createManagementClient } from '../../../utils/contentful-clients'
+import { assertLoggedIn, assertSpaceIdProvided } from '../../../utils/assertions'
+
+export const command = 'create'
+
+export const desc = 'Create an environment'
+
+export const builder = (yargs) => {
+  return yargs
+    .usage('Usage: contentful environment create --name \'Your Environment Name\'')
```
Comment (Khaledgarbaya) [inline lib/cmds/space_cmds/environment_cmds/create.js]: contentful **space** environment create --name ...

**Comment by Khaledgarbaya** [comment]
Test are all passing locally, this is probably an issue with Travis, there is something wrong with stdout

**Pull request merged**